specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2021-09-01/extensions.json

```diff
@@ -0,0 +1,864 @@
+          },
+          {
+            "name": "forceDelete",
+            "in": "query",
+            "description": "Delete the extension resource in Azure - not the normal asynchronous delete.",
+            "type": "boolean"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
+            "description": "Accepted.  The request has been accepted for processing."
+          },
+          "204": {
+            "description": "No Content.  The request has been accepted but the extension was not found."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      },
+      "patch": {
+        "tags": [
+          "Extensions",
+          "ClusterExtensions"
+        ],
+        "description": "Patch an existing Kubernetes Cluster Extension.",
+        "operationId": "Extensions_Update",
+        "x-ms-examples": {
+          "Update Extension": {
+            "$ref": "./examples/PatchExtension.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ExtensionNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "patchExtension",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/patchExtension"
+            },
+            "description": "Properties to Patch in an existing Extension."
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Request received successfully, and the resource will be updated asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/PatchResponse"
+            }
+          },
+          "409": {
```

Comment:
>409 [](http://example.com/codeflow?start=11&length=3)

What's the reason you are explicitly declaring a 409 instead of relying on the `default` error response? How does it change the generated SDK?

Reply:
This is for cases where a Patch request is already in progress and the client is issuing another one.  Since we store 'Updating' ProvisioningState in our DB (which gets updated to a terminal state after our Agent from the On-Prem cluster reports back on the previous Patch request), we do not want to allow a conflicting Patch request and so reject it and give 409 conflict as the reason for rejection.

Reply:
Sure, that's perfectly fine. I'm asking why it's explicitly listed. Your API can return all kinds of 4xx and 5xx codes (some of those you control and some come from ARM). None of them listed here, why is 409 special?

Reply:
(Btw, this isn't blocking the review necessarily - just trying to understand the scenario behind it.)

Reply:
Oh, I see what you mean.  I didn't realize it is not required to call out - however, this is the first time in my experience we are returning 409 and I thought it needs to be called out.